modules/client-handler/src/main/java/org/apache/ignite/client/handler/ClientInboundMessageHandler.java

```diff
@@ -993,9 +1024,11 @@ private long observableTimestamp(@Nullable ClientMessagePacker out) {
             if (meta instanceof HybridTimestamp) {
                 return ((HybridTimestamp) meta).longValue();
             }
+        } else {
+            return clockService.currentLong();
         }
 
-        return clockService.currentLong();
+        return HybridTimestamp.MIN_VALUE.longValue();
```

Comment:
I suggest rework a bit this method to see the code branching more clearly.
```java
private long observableTimestamp(@Nullable ClientMessagePacker out) {
        // Certain operations can override the timestamp and provide it in the meta object.
        if (out == null) {
            return clockService.currentLong();
        }

        Object meta = out.meta();

        if (meta instanceof HybridTimestamp) {
            return ((HybridTimestamp) meta).longValue();
        }

        return HybridTimestamp.MIN_VALUE.longValue();
    }
```

Reply:
Ok